index.bs

```diff
@@ -7090,10 +7090,10 @@ To <dfn>deserialize local value</dfn> given |local protocol value|, |realm| and
 1. If |local protocol value| matches the <code>script.ChannelValue</code> production,
    return [=create a channel=] with |session|, |realm| and |local protocol value|.
 
-1. Let |type| be the value of the <code>type</code> field of |local protocol value| or undefined if
+1. Let |type| be the value of the <code>type</code> field of |local protocol value| or null if
```

Comment:
If you're going to rewrite this, I'd start by rewriting to use the infra-based language we've adopted elsewhere e.g. start with something like `If |local protocol value| [=map/contains=] "<code>type</code>", let |type| be |local protocol value|["<code>type</code>"]. Otherwise let |type| be null.`

But actually reading the rest of the algorithm, it seems like it's always an error to not have "`type`" and I don't see a case that makes sense without `"value"`, so we could either return early if they're null or, if we're confident that the callers handle validation (e.g. all the data is coming from CDDL that enforces these values), we could assert that the fields exist.